cmd/cli/plugin/secret/registry_secret_add.go

```diff
@@ -185,3 +183,57 @@ func extractPassword() (string, error) {
 
 	return password, nil
 }
+
+func checkNamespaceList(kc kappclient.Client) error {
+	ns := &corev1.NamespaceList{}
+
+	err := kc.GetClient().List(context.Background(), ns)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkSecretExportExists(pkgClient tkgpackageclient.TKGPackageClient, kc kappclient.Client) error {
+	export := ""
+	secretExport, err := pkgClient.GetSecretExport(registrySecretOp)
+
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	// No error means we found a matching SecretExport
+
+	// Creating a SecretExport resource X that conflicts with a previously defined SecretExport Y that was exported to all namespaces could result in privilege escalation if the user does not access to other namespaces. This check prevents it by trying to list the namespaces
```

Comment:
@shyaamsn - Would you recommend a more terse comment? (I agree the statement can be confusing)

Reply:
I dont have a better recommendation either. Lets stick to this and iterate later if we have a better wording.